cpp/examples/strings/custom_with_malloc.cu

```diff
@@ -0,0 +1,150 @@
+
+  auto const name = d_names.element<cudf::string_view>(index);
+  auto const vis  = d_visibilities.element<cudf::string_view>(index);
+  if (vis == visible) {
+    auto const space_idx    = name.find(' ');
+    auto const first        = name.substr(0, space_idx);
+    auto const last_initial = name.substr(space_idx + 1, 1);
+    auto const output_size  = first.size_bytes() + last_initial.size_bytes() + 1;
+
+    char* output_ptr = static_cast<char*>(malloc(output_size));
+    d_output[index]  = cudf::string_view{output_ptr, output_size};
+
+    // build output string
+    memcpy(output_ptr, last_initial.data(), last_initial.size_bytes());
+    output_ptr += last_initial.size_bytes();
+    *output_ptr++ = ' ';
+    memcpy(output_ptr, first.data(), first.size_bytes());
+  } else {
+    d_output[index] = cudf::string_view{redaction.data(), redaction.size_bytes()};
+  }
+}
+
+/**
+ * @brief Frees the temporary individual string objects created in the
+ * redact_kernel
+ *
+ * Like malloc(), free() is not very efficient but must be called for
+ * each malloc() to return the memory to the CUDA malloc heap.
+ *
+ * @param redaction Redacted string replacement (not to be freed)
+ * @param d_output Output array of string_view objects to free
+ */
+__global__ void free_kernel(cudf::string_view redaction, cudf::string_view* d_output, int count)
+{
+  auto index = threadIdx.x + blockIdx.x * blockDim.x;
+  if (index >= count) return;
+
+  auto ptr = const_cast<char*>(d_output[index].data());
+  if (ptr != redaction.data()) free(ptr);
+}
+
+std::unique_ptr<cudf::column> redact_strings(cudf::column_view const& names,
+                                             cudf::column_view const& visibilities)
+{
+  auto stream = rmm::cuda_stream_default;
+  set_malloc_heap_size();
+
+  auto const d_names        = cudf::column_device_view::create(names, stream);
+  auto const d_visibilities = cudf::column_device_view::create(visibilities, stream);
+  auto const d_redaction    = cudf::string_scalar(std::string("X X"), true, stream);
+
+  constexpr auto block_size = 128;
+  auto const blocks         = (names.size() + block_size - 1) / block_size;
+
+  nvtxRangePushA("redact_strings");
+
+  // create a vector for the output strings' pointers
+  auto str_ptrs = new rmm::device_uvector<cudf::string_view>(names.size(), stream);
+
+  auto result = [&] {
+    // build the output strings
+    redact_kernel<<<blocks, block_size, 0, stream.value()>>>(
+      *d_names, *d_visibilities, d_redaction.value(), str_ptrs->data());
+    // create strings column from the string_view vector
+    // this copies all the individual strings into a single output column
+    return cudf::make_strings_column(*str_ptrs, cudf::string_view{nullptr, 0}, stream);
+  }();
+
+  // free the individual temporary memory pointers
+  free_kernel<<<blocks, block_size, 0, stream.value()>>>(
+    d_redaction.value(), str_ptrs->data(), names.size());
+  delete str_ptrs;
```

Comment:
Maybe a note about device_uvector() synchronizing the stream in it's destructor.

Reply:
It doesn't.